src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -62,8 +62,21 @@ public Type[] GetStartupTypes()
         public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
         {
             string binPath;
-            if (_extensionBundleManager.IsExtensionBundleConfigured())
+            var bindingsSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
You can avoid these allocations as we don't need those when bundles are not setup.